### PR TITLE
Run runtime against full driver-loader back compat

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#4329](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4329)

As a follow up to https://github.com/microsoft/FluidFramework/pull/15337 which added some testing to our real service e2e tests that run after build completion, we want to run the new runtime vs loader-driver compat (N-2 => LTS+1) against local and tinylicious e2e runs that run on our build - client packages. We are adding these tests to prevent loader-driver compat boundary regressions.

The internal build-client packages will fail if the full compat tests fail and thus create an alert for FF OCE's to investigate.

### CODE REVIEW

**Review by andre4i — APPROVED** [review]

File: tools/pipelines/build-client.yml [hunk 1]
```diff
@@ -119,6 +119,9 @@ extends:
     - ci:test:jest
     - ci:test:realsvc:local
     - ci:test:realsvc:tinylicious
+    - ${{ if ne(variables['System.TeamProject'], 'public') }}:
```
Comment (anthony-murphy) [inline tools/pipelines/build-client.yml:122]: isn't ci:test:realsvc:local:full a superset of ci:test:realsvc:local? can we avoid redundant work?

Comment (tyler-cai-microsoft) [reply tools/pipelines/build-client.yml:123]: Yeah, made an improvement
